### PR TITLE
Doc: Update readme.md

### PR DESCRIPTION
**Pull request by feng-intel**
## Type of Change

feature or bug fix or documentation or validation or others  
API changed or not

## Description

detail description

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: docs/source/3x/PT_FP8Quant.md [hunk 1]
```diff
@@ -75,30 +76,189 @@ Intel Neural Compressor provides general quantization APIs to leverage HPU FP8 c
...
+$ pip install -r requirements_lm_eval.txt  (Option)
+```
+### Check neural_compressor code
+> optimum-habana/examples/text-generation/utils.py
+>> initialize_model() -> setup_model() -> setup_quantization() -> FP8Config/prepare()/convert() 
+
+### FP8 KV cache
+Introduction: [kv-cache-quantization in huggingface transformers](https://huggingface.co/blog/kv-cache-quantization)    
+
+BF16 KVCache Code -> [Modeling_all_models.py -> KVCache()](https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/models/modeling_all_models.py#L40)
```
Comment (thuang6) [inline docs/source/3x/PT_FP8Quant.md:105]: not be good to link to certain line of code which is subject to change.  may explain some general idea or link to certain doc in OH

File: docs/source/3x/PT_FP8Quant.md [hunk 2]
```diff
@@ -75,30 +76,189 @@ Intel Neural Compressor provides general quantization APIs to leverage HPU FP8 c
...
+"meta-llama/Llama-3.1-8B-Instruct",    
+"Qwen/Qwen2.5-7B-Instruct", or     
+"mistralai/Mixtral-8x7B-Instruct-v0.1" and so on.    
+"--use_kv_cache" is to enable FP8 KV cache.
+
+### Profiling
+Add "--profiling_warmup_steps 5 --profiling_steps 2 --profiling_record_shapes" as args in the end of commandline of run_generation.py.     
+Refer to [torch.profiler.ProfilerActivity.HPU](https://github.com/huggingface/optimum-habana/blob/c9e1c23620618e2f260c92c46dfeb163545ec5ba/optimum/habana/utils.py#L305).    
+
+### FP8 Accuracy
```
Comment (thuang6) [inline docs/source/3x/PT_FP8Quant.md:139]: this acc data is not right and not update-to-date.  please check with @XuehaoSun for latest test result.  For Qwen2.5 model, there is bug opened, you can wait for that get fixed.

File: docs/source/3x/PT_FP8Quant.md [hunk 3]
```diff
@@ -75,30 +76,189 @@ Intel Neural Compressor provides general quantization APIs to leverage HPU FP8 c
...
+--request-rate 128 \
+--num-prompts 128 \
+--port 8080 \
+--sonnet-input-len 128 \
+--sonnet-output-len 128 \
+--sonnet-prefix-len 100
 ```
 
-## Examples
+### FP8 KV cache
```
Comment (thuang6) [inline docs/source/3x/PT_FP8Quant.md:254]: enable PatchedVLLMKVCache is not enough.  recent PR:  https://github.com/HabanaAI/vllm-fork/pull/569 enabled INC to patch ModuleFusedSDPA to FP8.  without this PR, there is no perf gain in attention from FP8 KV cache

File: docs/source/3x/PT_FP8Quant.md [hunk 4]
```diff
@@ -75,30 +76,189 @@ Intel Neural Compressor provides general quantization APIs to leverage HPU FP8 c
...
+"--use_kv_cache" is to enable FP8 KV cache.
+
+### Profiling
+Add "--profiling_warmup_steps 5 --profiling_steps 2 --profiling_record_shapes" as args in the end of commandline of run_generation.py.     
+Refer to [torch.profiler.ProfilerActivity.HPU](https://github.com/huggingface/optimum-habana/blob/c9e1c23620618e2f260c92c46dfeb163545ec5ba/optimum/habana/utils.py#L305).    
+
+### FP8 Accuracy 
+"lm_eval.tasks", "lm_eval.evaluator", "lm_eval" are installed from the above requirements_lm_eval.txt. The tasks can be set and the default is ["hellaswag", "lambada_openai", "piqa", "winogrande"], [more info](https://github.com/EleutherAI/lm-evaluation-harness/)    
+
+| `Llama-2-7b-hf`| fp8 & fp8 KVCache| bf16 w/o fp8 KVCache|
```
Comment (xin3he) [inline docs/source/3x/PT_FP8Quant.md:142]: bf16 w/o fp8 KVCache? -> `bf16 w/ bf16 KVCache` or `bf16 w/o bf16 KVCache`

**Review by xin3he — COMMENTED** [review]
`vllm_gaudi.png` is pretty good, I notice two typos.
- The model path in command is your local path. 
- `import neural_compressor.torch.quantization` is missing '.', same to vllm_hpu_extension

File: README.md [hunk 5]
```diff
@@ -33,51 +33,36 @@ support AMD CPU, ARM CPU, and NVidia GPU through ONNX Runtime with limited testi
...
-
-> **Note**:
-> There is a version mapping between Intel Neural Compressor and Gaudi Software Stack, please refer to this [table](./docs/source/3x/gaudi_version_map.md) and make sure to use a matched combination.
-
-#### Install torch/intel_extension_for_pytorch for Intel GPU
-https://intel.github.io/intel-extension-for-pytorch/index.html#installation
-
-#### Install torch for other platform
-https://pytorch.org/get-started/locally
+[Install intel_extension_for_pytorch for CPU -> Installation](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/)
```
Comment (chensuyue) [inline README.md:36]: Please remove ` -> Installation`, it looks not good.

File: README.md [hunk 6]
```diff
@@ -33,51 +33,36 @@ support AMD CPU, ARM CPU, and NVidia GPU through ONNX Runtime with limited testi
...
-> There is a version mapping between Intel Neural Compressor and Gaudi Software Stack, please refer to this [table](./docs/source/3x/gaudi_version_map.md) and make sure to use a matched combination.
-
-#### Install torch/intel_extension_for_pytorch for Intel GPU
-https://intel.github.io/intel-extension-for-pytorch/index.html#installation
-
-#### Install torch for other platform
-https://pytorch.org/get-started/locally
+[Install intel_extension_for_pytorch for CPU -> Installation](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/)    
+[Install intel_extension_for_pytorch for XPU -> Installation](https://intel.github.io/intel-extension-for-pytorch/xpu/latest/)    
+[Use Docker Image with torch installed for HPU](https://docs.habana.ai/en/latest/Installation_Guide/Bare_Metal_Fresh_OS.html#bare-metal-fresh-os-single-click)    --> **Note**: There is a version mapping between Intel Neural Compressor and Gaudi Software Stack, please refer to this [table](./docs/source/3x/gaudi_version_map.md) and make sure to use a matched combination.
```
Comment (chensuyue) [inline README.md:38]: Remove `--> `, put **Note** in a new line.

File: README.md [hunk 7]
```diff
@@ -33,51 +33,36 @@ support AMD CPU, ARM CPU, and NVidia GPU through ONNX Runtime with limited testi
...
-#### Install torch/intel_extension_for_pytorch for Intel GPU
-https://intel.github.io/intel-extension-for-pytorch/index.html#installation
-
-#### Install torch for other platform
-https://pytorch.org/get-started/locally
+[Install intel_extension_for_pytorch for CPU -> Installation](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/)    
+[Install intel_extension_for_pytorch for XPU -> Installation](https://intel.github.io/intel-extension-for-pytorch/xpu/latest/)    
+[Use Docker Image with torch installed for HPU](https://docs.habana.ai/en/latest/Installation_Guide/Bare_Metal_Fresh_OS.html#bare-metal-fresh-os-single-click)    --> **Note**: There is a version mapping between Intel Neural Compressor and Gaudi Software Stack, please refer to this [table](./docs/source/3x/gaudi_version_map.md) and make sure to use a matched combination.    
+[Install torch for other platform](https://pytorch.org/get-started/locally)    
+[Install Tensorflow](https://www.tensorflow.org/install)
```
Comment (chensuyue) [inline README.md:40]: tensorflow or TensorFlow.

File: README.md [hunk 8]
```diff
@@ -33,51 +33,36 @@ support AMD CPU, ARM CPU, and NVidia GPU through ONNX Runtime with limited testi
 
 ## Installation
 ### Install Framework
```
Comment (chensuyue) [inline README.md:35]: Maybe add some description for this session, like "Choose the necessary framework dependencies to install based on your deploy environment. "

File: README.md [hunk 9]
```diff
@@ -33,51 +33,36 @@ support AMD CPU, ARM CPU, and NVidia GPU through ONNX Runtime with limited testi
...
 # Install 2.X API + Framework extension API + TensorFlow dependency
 pip install neural-compressor[tf]
 ```
-> **Note**:
-> Further installation methods can be found under [Installation Guide](./docs/source/installation_guide.md). check out our [FAQ](./docs/source/faq.md) for more details.
+### Install Neural Compressor from Source for torch
+```
+git clone https://github.com/intel/neural-compressor.git
+cd neural-compressor
+pip install -r requirements.txt
```
Comment (chensuyue) [inline README.md:53]: For 3x torch, no need this line `pip install -r requirements.txt`.

**Review by xin3he — APPROVED** [review]
LGTM

File: README.md [hunk 10]
```diff
@@ -32,52 +32,31 @@ support AMD CPU, ARM CPU, and NVidia GPU through ONNX Runtime with limited testi
 * [2024/07] Performance optimizations and usability improvements on [client-side](./docs/source/3x/client_quant.md).
 
 ## Installation
+Choose the necessary framework dependencies to install based on your deploy environment.
 ### Install Framework
-#### Install torch for CPU
-```Shell
-pip install torch --index-url https://download.pytorch.org/whl/cpu
+[Install intel_extension_for_pytorch for CPU](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/)
```
Comment (chensuyue) [inline README.md:37]: Need to split those into several lines, make it clearer. 
image

Comment (chensuyue) [reply README.md:37]: - [Install intel_extension_for_pytorch for CPU](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/) 
- [Install intel_extension_for_pytorch for XPU](https://intel.github.io/intel-extension-for-pytorch/xpu/latest/) 
- [Use Docker Image with torch installed for HPU](https://docs.habana.ai/en/latest/Installation_Guide/Bare_Metal_Fresh_OS.html#bare-metal-fresh-os-single-click)
Note: There is a version mapping between Intel Neural Compressor and Gaudi Software Stack, please refer to this [table](https://github.com/intel/neural-compressor/blob/e0b65444535a3ce502b2cf5c65921dfa29786dd0/docs/source/3x/gaudi_version_map.md) and make sure to use a matched combination.
- [Install torch for other platform](https://pytorch.org/get-started/locally)
- [Install tensorflow](https://www.tensorflow.org/install)

File: docs/source/3x/PT_FP8Quant.md [hunk 11]
```diff
@@ -75,30 +76,191 @@ Intel Neural Compressor provides general quantization APIs to leverage HPU FP8 c
...
+"--use_kv_cache" is to enable FP8 KV cache.
+
+### Profiling
+Add "--profiling_warmup_steps 5 --profiling_steps 2 --profiling_record_shapes" as args in the end of commandline of run_generation.py.     
+Refer to [torch.profiler.ProfilerActivity.HPU](https://github.com/huggingface/optimum-habana/blob/c9e1c23620618e2f260c92c46dfeb163545ec5ba/optimum/habana/utils.py#L305).    
+
+### FP8 Accuracy 
+"lm_eval.tasks", "lm_eval.evaluator", "lm_eval" are installed from the above requirements_lm_eval.txt. The tasks can be set and the default is ["hellaswag", "lambada_openai", "piqa", "winogrande"], [more info](https://github.com/EleutherAI/lm-evaluation-harness/)    
+
+| `Llama-2-7b-hf`| fp8 & fp8 KVCache| bf16 w/ fp16 KVCache|
```
Comment (xin3he) [inline docs/source/3x/PT_FP8Quant.md:144]:
```suggestion
| `Llama-2-7b-hf`| fp8 & fp8 KVCache| bf16 w/ bf16 KVCache|
```

File: docs/source/3x/PT_FP8Quant.md [hunk 12]
```diff
@@ -75,30 +76,191 @@ Intel Neural Compressor provides general quantization APIs to leverage HPU FP8 c
...
+|---------------|---------|--------|
+| hellaswag     | 0.5691097390957977   | 0.5704043019318861    |
+| lambada_openai| 0.7360760721909567   | 0.7372404424607025 | 
+| piqa          | 0.7850924918389554   | 0.7818280739934712 |
+| winogrande    | 0.6929755327545383   | 0.6929755327545383 |
+
+| `Qwen2.5-7B-Instruct`| fp8 & fp8 KVCache| bf16 w/ fp16 KVCache|
```
Comment (xin3he) [inline docs/source/3x/PT_FP8Quant.md:151]: same

File: docs/source/3x/PT_FP8Quant.md [hunk 13]
```diff
@@ -75,30 +76,191 @@ Intel Neural Compressor provides general quantization APIs to leverage HPU FP8 c
...
+|---------------|---------|--------|
+| hellaswag     |  0.2539334793865764  |   0.2539334793865764    |
+| lambada_openai| 0.0   | 0.0 | 
+| piqa          | 0.5391730141458106   | 0.5391730141458106 |
+| winogrande    | 0.4956590370955012  | 0.4956590370955012 |
+
+| `Llama-3.1-8B-Instruct`| fp8 & fp8 KVCache| bf16 w/ fp16 KVCache|
```
Comment (xin3he) [inline docs/source/3x/PT_FP8Quant.md:158]: same

File: docs/source/3x/PT_FP8Quant.md [hunk 14]
```diff
@@ -75,30 +76,191 @@ Intel Neural Compressor provides general quantization APIs to leverage HPU FP8 c
...
+|---------------|---------|--------|
+| hellaswag     | 0.5934076877116112   |   0.5975901214897431    |
+| lambada_openai| 0.7230739375121289   | 0.7255967397632447 | 
+| piqa          | 0.7932535364526659   | 0.8030467899891186 |
+| winogrande    | 0.7434885556432518  | 0.7371744277821626 |
+
+
+| `Mixtral-8x7B-Instruct-v0.1`| fp8 & fp8 KVCache| bf16 w/ fp16 KVCache||
```
Comment (xin3he) [inline docs/source/3x/PT_FP8Quant.md:166]: same

**Review by xin3he — CHANGES_REQUESTED** [review]

**Review by chensuyue — APPROVED** [review]

**Review by xin3he — APPROVED** [review]
